src/boot/efi/boot.c

```diff
@@ -1453,8 +1453,30 @@ static BOOLEAN config_entry_add_loader_auto(Config *config, EFI_HANDLE *device,
                 return TRUE;
 
         /* do not add an entry for ourselves */
-        if (loaded_image_path && StriCmp(loader, loaded_image_path) == 0)
-                return FALSE;
+        if (loaded_image_path) {
+                if (StriCmp(loader, loaded_image_path) == 0)
+                        return FALSE;
+                else {
+                        UINTN lenA;
+                        UINTN lenB;
+                        INTN equal = 1;
+                        CHAR8 *contentA = NULL;
+                        CHAR8 *contentB = NULL;
+
+                        err = file_read(root_dir, loader, 0, 0, &contentA, &lenA);
+                        if (!EFI_ERROR(err))
+                                err = file_read(root_dir, loaded_image_path, 0, 0, &contentB, &lenB);
```

Comment:
hmpf, this is potentially huuugge, no? is this really a good idea? 

what precisely are you trying to fix here?

Reply:
This is in the situation where we boot as "systemd/systemd-bootx64.efi" as created by bootctl for example. If the backup bootloader in "boot/bootx64.efi" is systemd-boot as well, we would end up putting ourselves to the list for no benefit.

I highly doubt anyone will ever put such a huge bootloader that it would become a performance bottleneck. In case there are errors, the code will just assume the loaders are different and continue.

Reply:
well, somebody could also have install a full kernel with initrd as bootx64.efi, no? it sounds a bit brutal to read all that in just to throw it away afterwards, no?